src/ninja.cc

```diff
@@ -1059,34 +1062,43 @@ bool WarningEnable(const string& name, Options* options) {
 "  phonycycle={err,warn}  phony build statement references itself\n"
     );
     return false;
-  } else if (name == "dupbuild=err") {
+  }
+
+  if (name == "dupbuild=err") {
```

Comment:
I don't see replacing if-else's with stand-alone if's as an improvement.

Reply:
If you really wanted to do that, consider instead making an enum, for each string contents allowed, and then make a function that does
```
enum myenum string_to_my_enum(std::string const& str)
{
    if(str == "string_1") return enum_1;
    if(str == "string_2") return enum_2;
    if(str == "string_3") return enum_3;
    if(str == "string_4") return enum_4;
    return enum_unknown;
}
```

and then use a switch-case in this function.